controllers/etcdrestore_controller.go

```diff
@@ -0,0 +1,524 @@
+	log := r.Log.WithValues("etcdrestore", req.NamespacedName)
+
+	log.Info("Begin restore reconcile")
+
+	var restore etcdv1alpha1.EtcdRestore
+	if err := r.Get(ctx, req.NamespacedName, &restore); err != nil {
+		// Can't find the resource. We could have just been deleted? If so, no need to do anything as the owner
+		// references will clean everything up with a cascading delete.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	} else {
+		log.Info("Found restore object in Kubernetes, continuing")
+		// Found the resource, continue with reconciliation.
+	}
+
+	if restore.Status.Phase == etcdv1alpha1.EtcdRestorePhaseCompleted ||
+		restore.Status.Phase == etcdv1alpha1.EtcdRestorePhaseFailed {
+		// Do nothing here. We're already finished.
+		log.Info("Phase is set to an end state. Taking no further action", "phase", restore.Status.Phase)
+		return ctrl.Result{}, nil
+	} else if restore.Status.Phase == etcdv1alpha1.EtcdRestorePhasePending {
+		// In any other state continue with the reconciliation. In particular we don't *read* the other possible states
+		// as once we know that we need to reconcile at all we will use the observed sate of the cluster and not our own
+		// status field.
+		log.Info("Phase is not set to an end state. Continuing reconciliation.", "phase", restore.Status.Phase)
+	} else {
+		log.Info("Restore has no phase, setting to pending")
+		restore.Status.Phase = etcdv1alpha1.EtcdRestorePhasePending
+		err := r.Client.Status().Update(ctx, &restore)
+		return ctrl.Result{}, err
+	}
+
+	// Simulate the cluster, peer, and PVC we'll create. We won't ever *create* any of these other than the PVC, but
+	// we need them to generate the PVC using the same code-path that the main operator does. Yes, in theory if you
+	// upgraded the operator part-way through a restore things could get super fun if the way of making these changed.
+	expectedCluster := clusterForRestore(restore)
+	expectedCluster.Default()
+	// We know that the cluster will be of size one for the restore. So only consider peer zero.
+	expectedPeerNames := expectedPeerNamesForCluster(expectedCluster)
+	expectedPeers := make([]etcdv1alpha1.EtcdPeer, len(expectedPeerNames))
+	for i, peerName := range expectedPeerNames {
+		expectedPeers[i] = *peerForCluster(expectedCluster, peerName)
+		expectedPeers[i].Default()
+		configurePeerBootstrap(&expectedPeers[i], expectedCluster)
+	}
+	expectedPVCs := make([]corev1.PersistentVolumeClaim, len(expectedPeers))
+	for i, peer := range expectedPeers {
+		expectedPVCs[i] = *pvcForPeer(&peer)
+		markPVC(restore, &expectedPVCs[i])
+	}
+
+	// Step 1. Create the PVCs.
+	for _, expectedPVC := range expectedPVCs {
+		var pvc corev1.PersistentVolumeClaim
+		err := r.Get(ctx, name(&expectedPVC), &pvc)
+
+		if apierrors.IsNotFound(err) {
+			// No PVC. Make it!
+			log.Info("Creating PVC", "pvc-name", name(&expectedPVC))
+			err := r.Create(ctx, &expectedPVC)
+			return ctrl.Result{}, err
+		} else if err != nil {
```

Comment:
This needn't be in an `else` branch since the previous block returns. Consider simplifying this statement.